enterprise/core-edge/src/main/java/org/neo4j/coreedge/raft/log/segmented/DumpSegmentedRaftLog.java

```diff
@@ -53,34 +53,45 @@ private int dump( String filenameOrDirectory, PrintStream out )
             throws IOException, DamagedLogStorageException, DisposedException
     {
         LogProvider logProvider = NullLogProvider.getInstance();
-        int logsFound = 0;
+        final int[] logsFound = {0};
+        FileNames fileNames = new FileNames( new File( filenameOrDirectory ) );
+        ReaderPool readerPool = new ReaderPool( 0, logProvider, fileNames, fileSystem, systemUTC() );
         RecoveryProtocol recoveryProtocol =
-                new RecoveryProtocol( fileSystem, new FileNames( new File( filenameOrDirectory ) ), marshal,
-                        logProvider );
+                new RecoveryProtocol( fileSystem, fileNames, readerPool, marshal, logProvider );
         Segments segments = recoveryProtocol.run().segments;
 
-        ListIterator<SegmentFile> segmentFileIterator = segments.getSegmentFileIteratorAtStart();
+        segments.visit( (segment) -> {
+                logsFound[0]++;
```

Comment:
Can't we make `visit` return the number of visited logs rather than this array hack?